pkg/cloudprovider/fake/instancetype.go

```diff
@@ -23,106 +25,124 @@ import (
+	if options.CPU.IsZero() {
+		options.CPU = resource.MustParse("4")
 	}
-	if options.memory.IsZero() {
-		options.memory = resource.MustParse("4Gi")
+	if options.Memory.IsZero() {
+		options.Memory = resource.MustParse("4Gi")
 	}
-	if options.pods.IsZero() {
-		options.pods = resource.MustParse("5")
+	if options.Pods.IsZero() {
+		options.Pods = resource.MustParse("5")
 	}
 	return &InstanceType{
-		InstanceTypeOptions: InstanceTypeOptions{
-			name:             options.name,
-			offerings:        options.offerings,
-			architecture:     options.architecture,
-			operatingSystems: options.operatingSystems,
-			cpu:              options.cpu,
-			memory:           options.memory,
-			pods:             options.pods,
-			nvidiaGPUs:       options.nvidiaGPUs,
-			amdGPUs:          options.amdGPUs,
-			awsNeurons:       options.awsNeurons,
-			awsPodENI:        options.awsPodENI,
+		options: InstanceTypeOptions{
+			Name:             options.Name,
+			Offerings:        options.Offerings,
+			Architecture:     options.Architecture,
+			OperatingSystems: options.OperatingSystems,
+			CPU:              options.CPU,
+			Memory:           options.Memory,
+			Pods:             options.Pods,
+			NvidiaGPUs:       options.NvidiaGPUs,
+			AMDGPUs:          options.AMDGPUs,
+			AWSNeurons:       options.AWSNeurons,
+			AWSPodENI:        options.AWSPodENI,
 		},
 	}
 }
 
+// InstanceTypes creates instance types with incrementing resources
+// 2Gi of RAM and 10 pods for every 1vcpu
+// i.e. 1vcpu, 2Gi mem, 10 pods
+//      2vcpu, 4Gi mem, 20 pods
+//      3vcpu, 6Gi mem, 30 pods
+func InstanceTypes(total int) []cloudprovider.InstanceType {
+	instanceTypes := []cloudprovider.InstanceType{}
+	for i := 0; i < total; i++ {
+		instanceTypes = append(instanceTypes, NewInstanceType(InstanceTypeOptions{
+			Name:   fmt.Sprintf("fake-it-%d", i),
+			CPU:    resource.MustParse(fmt.Sprintf("%d", i+1)),
+			Memory: resource.MustParse(fmt.Sprintf("%dGi", (i+1)*2)),
+			Pods:   resource.MustParse(fmt.Sprintf("%d", (i+1)*10)),
+		}))
+	}
+	return instanceTypes
+}
+
 type InstanceTypeOptions struct {
-	name             string
-	offerings        []cloudprovider.Offering
-	architecture     string
-	operatingSystems sets.String
-	cpu              resource.Quantity
-	memory           resource.Quantity
-	pods             resource.Quantity
-	nvidiaGPUs       resource.Quantity
-	amdGPUs          resource.Quantity
-	awsNeurons       resource.Quantity
-	awsPodENI        resource.Quantity
+	Name             string
+	Offerings        []cloudprovider.Offering
+	Architecture     string
+	OperatingSystems sets.String
+	CPU              resource.Quantity
+	Memory           resource.Quantity
+	Pods             resource.Quantity
+	NvidiaGPUs       resource.Quantity
+	AMDGPUs          resource.Quantity
+	AWSNeurons       resource.Quantity
+	AWSPodENI        resource.Quantity
 }
 
 type InstanceType struct {
-	InstanceTypeOptions
+	options InstanceTypeOptions
```

Comment:
Too fancy?

Reply:
This was necessary because the funcs defined in the Interface now match the struct fields of InstanceTypeOptions, since they are now exported.

Reply:
Hmm -- I actually kind of did this intentionally to encourage not making them public. Why did you need to make them public?

Reply:
Making them public allows for more isolated tests with cloud provider specific data. For example, in my testing of the binpacking pkg, I loaded all the real AWS InstanceTypes into fake instance types and ran that thru the packing logic for lightweight,  real-world testing.  

I'm more in favor of keeping them public since it makes loading arbitrary instance types easier.